site/content/docs/getting-started/03-containers/index.md

```diff
@@ -0,0 +1,241 @@
+
+But what if we want to add another item to our box, like maybe an image? Let's see what happens if we place an image in the content box with our text:
+
+{% rustdoc_test() %}
+
+```rust
+# use raui::prelude::*;
+pub fn app(_ctx: WidgetContext) -> WidgetNode {
+    // Create our text box properties
+    // ...
+#    let text_box_props = Props::new(TextBoxProps {
+#        text: "Hello World!".into(),
+#        color: Color {
+#            r: 0.0,
+#            g: 0.0,
+#            b: 0.0,
+#            a: 1.0,
+#        },
+#        font: TextBoxFont {
+#            // We specify the path to our font
+#            name: "resources/verdana.ttf".to_owned(),
+#            size: 60.0,
+#        },
+#        // Use the defaults for the rest of the text box settings
+#        ..Default::default()
+#    })
+#    .with(ContentBoxItemLayout {
+#        // Specify a margin of 10 on every side
+#        margin: Rect {
+#            left: 10.,
+#            right: 10.,
+#            top: 10.,
+#            bottom: 10.,
+#        },
+#        // Use the default value for the rest of the layout options
+#        ..Default::default()
+#    });
+
+    // Create the props for our image
+    let image_box_props = Props::new(ImageBoxProps {
+        // The material defines what image or color to use for the box
+        material: ImageBoxMaterial::Image(ImageBoxImage {
+            // The path to our image
+            id: "resources/cats.jpg".to_owned(),
+            ..Default::default()
+        }),
+        // Have the image fill it's container
+        width: ImageBoxSizeValue::Fill,
+        height: ImageBoxSizeValue::Fill,
+        ..Default::default()
+    });
+
+    widget! {
+        (content_box [
+            // Add our image box above our text box, that will make sure that the
+            // image appears _under_ our text box when the UI is rendered
+            (image_box: {image_box_props})
+            (text_box: {text_box_props})
+        ])
+    }
+}
+```
+
+{% end %}
+
+![text and image](text-and-image.png)
+
+> **Note:** You can download the demo cat image [here](https://github.com/PsichiX/raui/raw/master/site/rust/guide_03/resources/cats.jpg).
+
+Notice that the content box didn't make sure the was any "room" for the text or the image to sit side-by-side with each-other, it just stacked them right on top of each-other. Also
+
+If we wanted to have them line up without overlapping we would use a [`flex_box`].
+
+[`flex_box`]: https://docs.rs/raui/latest/raui/core/widget/component/containers/flex_box/fn.flex_box.html
+
+## Flex Box
+
+Flex boxes in RAUI are somewhat similar to [CSS flexboxes][css_flexboxes], but they aren't necessarily going to work the same in all cases. Flex boxes lay things out in a row or a column depending on whether or not their _direction_. For convenience, RAUI has [`vertical_box`] and [`horizontal_box`] components that are simply [`flex_box`] component with their direction set to vertical or horizontal by default.
```

Comment:
the thing with flex box in RAUI is that it might seems to be exact the same as CSS flexbox but it was purposefully modified. After the review i'm going to investigate the margin and grow/shrink problem you've found in the other issue, if they are bugs they are gonna be fixed, but RAUI flex box won't be exactly same as CSS flex box bc RAUI flex box is focused on single pass size calculations which brings some edge cases but doesn't produce performance problems with multipass box size calculations as solutions that strictly follow CSS flex box - after the issue investigation i'll write a description of RAUI version of flex box and it's rules to clarify misconceptions (this is also the case where now i honestly regret calling it a flex box, because it might bring wrong assumptions 🤔) so expect some commits from me to this PR :p

Reply:
That makes sense. I wasn't sure how much it was intended to match CSS or not so I didn't want to say it shoud be pretty close if it wasn't supposed to be, but I'd feel comfortable saying that it should be the same in most situations in that case.

> so expect some commits from me to this PR :D

I actually opened the PR from my organization account so it doesn't have that "allow maintainers to push commits" feature. You could create a separate branch that I could merge into this branch, or we could just merge your changes after  merging this PR maybe?

Reply:
> or we could just merge your changes after merging this PR maybe?
that works for me!

Reply:
I updated the comment just to say that they are similar to CSS Flexboxes, and added a note that links to a new ( blank ) "Layout in Depth" page. You can use that section to write about how layout works in more detail. Feel free to rename the page to whatever you want.

I think that means we can merge this and then you can push whatever commits extra you want to `next` for the guide.